File: api/Domain/Model/Employee.cs
namespace Api.Domain.Model;

/// <summary>
/// Core data model for an employee.
/// </summary>
public class Employee : MongoEntityBase
{
    /// <summary>
    /// The first name of the employee.
    /// </summary>
    public string? FirstName { get; set; }

    /// <summary>
    /// The last name of the employee.
    /// </summary>
    public string? LastName { get; set; }

    /// <summary>
    /// The salary assigned to the employee.
    /// </summary>
    public int Salary { get; set; } = 0;

    /// <summary>
    /// A reference to the Company that an Employee works for.
    /// </summary>
    [MongoRef(typeof(Company))]
    public EntityRef? Company { get; set; }
}

File: web/src/services/models/Employee.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { EntityRef } from './EntityRef';

/**
 * Core data model for an employee.
 */
export type Employee = {
    /**
     * The ID of the entity in the MongoDB
     */
    id?: string | null;
    /**
     * The name associated with the entity in the MongoDB
     */
    label?: string | null;
    /**
     * The first name of the employee.
     */
    firstName?: string | null;
    /**
     * The last name of the employee.
     */
    lastName?: string | null;
    /**
     * The salary assigned to the employee.
     */
    salary?: number;
    company?: EntityRef;
}
